cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpInstanceResourceBuilder.java

```diff
@@ -230,26 +230,36 @@ private void assignToExistingInstanceGroup(GcpContext context,
         String projectId = context.getProjectId();
         String zone = context.getLocation().getAvailabilityZone().value();
 
-        List<CloudResource> instanceGroupResources = filterGroupFromName(filterResourcesByType(context.getGroupResources(group.getName()),
-                ResourceType.GCP_INSTANCE_GROUP), group.getName());
-
-        if (!instanceGroupResources.isEmpty() && doesGcpInstanceGroupExist(compute, projectId, zone, instanceGroupResources.get(0))) {
-            LOGGER.info("adding instance {} to group {} in project {}", instance.getName(), group.getName(), projectId);
-            InstanceGroupsAddInstancesRequest request = createAddInstancesRequest(instance, projectId, zone);
-            AddInstances addInstances = compute.instanceGroups().addInstances(projectId,
-                    zone, instanceGroupResources.get(0).getName(), request);
-            try {
-                Operation execute = addInstances.execute();
-                verifyOperation(execute, buildableResource);
-            } catch (GoogleJsonResponseException e) {
-                LOGGER.error("Error in Google response, unable to add instance {} to group {} : {} for {}",
-                        instance.getName(), group.getName(), e.getMessage(), e.getDetails().getMessage());
-                throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
-            }
-        } else {
+        List<CloudResource> instanceGroups = filterInstanceGroupsByName(context.getGroupResources(group.getName()), group.getName());
+
+        if (instanceGroups.isEmpty()) {
+            LOGGER.info("skipping group assignment {} was not found in project {}", group.getName(), projectId);
+            return;
+        }
+        if (instanceGroups.size() > 1) {
+            LOGGER.warn("Found multiple instance groups to assign, trying to find an active one. Group {}, project {}", group.getName(), projectId);
```

Comment:
why is it warn? I think info should be sufficient as it doesn't seem like an issue here. maybe logging the instancegroups always here would help to identify issues later ( I mean you might remove the if condition and always log)

Reply:
changed to _info_ . I made it _warn_ initially because having multiple instance group after filtering is not a normal situation. Previous implementation didn't care about that and just took the first item from the list which I find quite brave in a bad sense. But since it worked before without a warning it can work now without it too.

Reply:
Reverted the code part back to master.